src/jit/compiler.cpp

```diff
@@ -540,7 +540,7 @@ var_types    Compiler::argOrReturnTypeForStruct(unsigned size, CORINFO_CLASS_HAN
         break;
 
     default:
-#if FEATURE_MULTIREG_RET
+#if FEATURE_MULTIREG_RET && !defined(_TARGET_X86_)
```

Comment:
This is a bit awkward.  I think that this method should be refactored, as it is now quite difficult to follow the logic. Perhaps the 32-bit architectures could at least be separated from the 64-bit ones.